apps/sms/views/shared/js/navigation.js

```diff
@@ -1,361 +1,921 @@
-/* global Promise,
-      EventDispatcher
-*/
-
-/* exported Navigation */
+/*global
+  EventDispatcher,
+  Utils
+ */
 
+(function(window) {
 'use strict';
 
-(function(window) {
+/* The factory will be used in unit tests. */
+var NavigationFactory = window.NavigationFactory = function(window) {
+
+/**
+ * The Defer type is useful when creating promises.
+ * @typedef {Object} Defer
+ * @property {function(*)} resolve The Promise's resolve function.
+ * @property {function(*)} reject The Promise's reject function.
+ * @property {Promise} promise The actual promise.
+ */
+
+/**
+ * @function
+ * The debug function translates to console.log in debug mode.
+ *
+ * @param {String} arg1 The first value to be displayed to the console.
+ * @param {...*} args The other values to be displayed to the console, or the
+ * parameters to the first string.
+ */
+var debug = 1 ?
+  (arg1, ...args) => console.log('[Navigation] ' + arg1, ...args) :
+  () => {};
+
+/**
+  * This contains an array of view characteristics, necessary to handle
+  * navigation requests.
+  * Each item has the following properties:
+  * - name: this is the name of the view. This will be used also in the URL
+  *   hash if the same HTML file holds several views.
+  * - url: this is the URL used to match the current location and find the
+  *   current view.
+  * - view: this is the actual JavaScript object that manages this view.
+  * - previous (optional): this is the name of the previous view; this is
+  *   necessary to decide if we do a forward/backward animation.
+  * - partOf (optional): this is used when a view is part of another view's
+  *   HTML document. In that case one view is "master" and the other are only
+  *   part of the master view.
+  * - alsoContains (optional): this is the opposite and holds the view's
```

Comment:
__optional nit:__ maybe we don't need to add something that isn't used - or you have plans to use any time soon?